### PR TITLE
Fix several aspects of the "disconnected" dialog

### PR DESCRIPTION
**Pull request by huffman**
Changes:

 1. On first load, wait longer to show the disconnected dialog
 1. If trying to go to a place that requires login, don't show the can't connect dialog
 1. Make the can't connect dialog go away if you get connected to the domain

Note: The disconnected dialog is the one that says "No Connection" at the top

To test:
 1. Go to your home domain, ensure that the disconnected dialog does not show
 1. Shut your local server down, ensure that the disconnected dialog comes up
 1. Manually close the dialog, start the server back up
 1. Shut your local server down, ensure that the disconnected dialog comes up
 1. DO NOT close the dialog, start the server back up, ensure that the dialog goes away automatically
 1. Shut your local server down, ensure that the disconnected dialog comes up
 1. Restart interface, ensure that the disconnected dialog comes up after more than 10 seconds
 1. Close the dialog
 1. Ensure you are logged out. Go to `sport`
 1. Ensure that the "login" dialog comes up but that the disconnected dialog does __not__
 1. Go to a domain running a different protocol version (`welcome`, for instance)
 1. Ensure the packet version mismatch pops up, but the disconnected dialog does __not__

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: e7023b3d99a09f6f43856f13eb4dd0ad35ea284e available at: https://gist.github.com/268aa399a89e61d95aa79d936a98a0b9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8891-e7023b3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8891-e7023b3.dmg

**Comment by ctrlaltdavid** [comment]
Tested on Windows 10. Worked per test plan.

**Comment by huffman** [comment]
@ctrlaltdavid Thanks! I had to make a small change, I added steps 11 and 12. Could you test those as well?

File: interface/src/ConnectionMonitor.cpp [hunk 1]
```diff
@@ -18,29 +18,40 @@
...
+        qDebug() << "ConnectionMonitor: Showing connection failure window";
+        DependencyManager::get<DialogsManager>()->setDomainConnectionFailureVisibility(true);
+    });
 }
 
-void ConnectionMonitor::disconnectedFromDomain() {
+void ConnectionMonitor::startTimer() {
+    qDebug() << "ConnectionMonitor: Starting timer";
+    _timer.setInterval(DISPLAY_AFTER_DISCONNECTED_FOR_X_MS);
     _timer.start();
```
Comment (Atlante45) [inline interface/src/ConnectionMonitor.cpp]: Not blocking, but you could just do `_time.start(DISPLAY_AFTER_DISCONNECTED_FOR_X_MS);` here.

File: interface/src/ui/DialogsManager.cpp [hunk 2]
```diff
@@ -63,6 +64,14 @@ void DialogsManager::indicateDomainConnectionFailure() {
     OffscreenUi::information("No Connection", "Unable to connect to this domain. Click the 'GO TO' button on the toolbar to visit another domain.");
 }
```
Comment (Atlante45) [inline interface/src/ui/DialogsManager.cpp]: You can remove this function now.

**Comment by hifi-gustavo** [comment]
Build console output for commit: f3ef0cebedee2504c18d9e429436a2208e0beea2 available at: https://gist.github.com/268aa399a89e61d95aa79d936a98a0b9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8891-f3ef0ce.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8891-f3ef0ce.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e0c109d6cab94e004c5de2ad0fc23421564e525b available at: https://gist.github.com/268aa399a89e61d95aa79d936a98a0b9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8891-e0c109d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8891-e0c109d.dmg

**Comment by ctrlaltdavid** [comment]
Retested on Windows, including the extra steps. Worked per test plan.

**Pull request merged**